app/knight/knight.py

```diff
@@ -0,0 +1,62 @@
+from app.knight_equipment.armour import Armour
+from app.knight_equipment.potion import Potion
+from app.knight_equipment.weapon import Weapon
+
+
+class Knight:
+    def __init__(self, knight_config: dict) -> None:
```

Comment:
How about make init method get name, power, hp... one by one here in parameters, not whole knight_config dict?
It will make easier to understand required parameters to create a Knight without looking directly in this method
It'll also let you set default values easier

Reply:
After that you could use unpacking (using `**`) to create knights + list comprehension to create list of knights
`[Knight(**knight) for knight in knight_config]`